src/main/java/controllers/FeedController.java

```diff
@@ -30,56 +29,171 @@ public FeedController(InOut inOut, LoginManager loginManager) {
         }
     }
 
+    private String generateFeedActions(ArrayList<Post> posts) {
+        this.postsActionMap = new HashMap<>();
+        StringBuilder postsString = new StringBuilder("Enter a post number for a detailed view of that post: ");
+
+        for (int i=0; i < posts.size(); i++) {
+            this.postsActionMap.put(i, posts.get(i));
+            postsString.append("\n");
+            postsString.append(i).append(" ");
+            // Change getAuthorId to getAuthor
+            postsString.append(posts.get(i).getAuthorId());
+            postsString.append("'s ").append(posts.get(i).getRecipe().getTitle());
+        }
+        return postsString.toString();
+    }
+
+    private void selectOnePost(String postsString, int numOfDisplayedPosts) {
+        int postNumber = this.getSelectedPostInput(postsString, numOfDisplayedPosts);
+        Post selectedPost = this.postsActionMap.get(postNumber);
+        this.displayPostInfo(selectedPost);
+
+        int postAction = this.getPostActionInput();
+
+        this.runPostAction(selectedPost, postAction);
+        this.displayPostInfo(selectedPost);
+    }
+
+    private int getSelectedPostInput(String postsString, int numOfDisplayedPosts) {
+        int postNumber = 0;
+        boolean success = false;
+        while (!success) {
+            try {
+                do {
+                    String postSelection = this.inOut.getInput(postsString);
+                    postNumber = Integer.parseInt(postSelection);
+                } while (postNumber < 0 || postNumber > numOfDisplayedPosts - 1);
+                success = true;
+            } catch (IOException e) {
+                this.inOut.setOutput("An error occurred: " + e);
+            } catch (NumberFormatException nfe) {
+                this.inOut.setOutput("You entered an invalid action input.");
+            }
+        }
+        return postNumber;
+    }
+
+    private int getPostActionInput() {
+        int postAction = 0;
+        boolean success = false;
+        while (!success) {
+            try {
+                do {
+                    String postActionPrompt = "Please select an action for this post: \n"
+                            + "0 Like the Post \n" + "1 Comment on the Post";
+                    String postActionString = this.inOut.getInput(postActionPrompt);
+                    postAction = Integer.parseInt(postActionString);
+                } while (postAction < 0 || postAction > 1);
+                success = true;
+            } catch (IOException e) {
+                this.inOut.setOutput("An error occurred: " + e);
+            } catch (NumberFormatException nfe) {
+                this.inOut.setOutput("You entered an invalid action input.");
+            }
+        }
+        return postAction;
+    }
+
+    private Comment getUserComment() {
+        // This function MAY BE REMOVED if it is implemented in PostController instead.
```

Comment:
Please discuss with whoever is implementing PostController and decide to remove or keep.